compat/mingw.c

```diff
@@ -3587,8 +3587,29 @@ static void setup_windows_environment(void)
 			struct strbuf buf = STRBUF_INIT;
 			strbuf_addstr(&buf, tmp);
 			if ((tmp = getenv("HOMEPATH"))) {
+				/*
+				 * Since Vista, HOMEDRIVE/HOMEPATH are set to Windows\System32
+				 * for secondary logons/services instead of being unset.
+				 * We don't want to (and likely can't) save stuff there
+				 * so ignore them in such a case.
+				 */
+				wchar_t w_system_dir[MAX_PATH];
+				wchar_t *w_buf;
+				size_t len_w_buf;
 				strbuf_addstr(&buf, tmp);
-				if (is_directory(buf.buf))
+				if (!(GetSystemDirectoryW(
+						(LPWSTR)w_system_dir,
```

Comment:
Isn't this cast totally unnecessary? And if it is, it would be a lot nicer to avoid that awkward line break after an opening parenthesis.

Reply:
Maybe we should use `WCHAR` instead of `wchar_t`, then? Would that fix it?